### PR TITLE
Adds the simpleEnum class back.

### PR DESCRIPTION
**Pull request by cheqianh**
*Description of changes:*
* Adds its meta class back without using six.
* Adds unit tests back
* Marks it as deprecation. Adds deprecation warnings for the class.

*Issue #, if available:*
https://github.com/amazon-ion/ion-python/issues/241

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: amazon/ion/util.py [hunk 1]
```diff
@@ -217,15 +226,106 @@ def bit_length(value):
...
+        """Looks up an enumeration value field by integer value."""
+        return cls._enum_members[name]
+
+    def __iter__(self):
+        """Iterates through the values of the enumeration in no specific order."""
+        return iter(self._enum_members.values())
+
+
+class Enum(int, metaclass=_EnumMetaClass):
+    """Simple integer based enumeration type.
```
Comment (popematt) [inline amazon/ion/util.py:274]:
```suggestion
    """This is a deprecated, internal-only class in ion-python—do NOT use it for any reason.
    
    Simple integer based enumeration type.
```

File: amazon/ion/util.py [hunk 2]
```diff
@@ -217,15 +226,106 @@ def bit_length(value):
...
     def total_seconds(td):
         return td.total_seconds()
 
-
 bit_length.__doc__ = 'Returns the bit length of an integer'
 total_seconds.__doc__ = 'Timedelta ``total_seconds`` with backported support in Python 2.6'
+
+
+class _EnumMetaClass(type):
+    """Metaclass for simple enumerations.
```
Comment (popematt) [inline amazon/ion/util.py:243]: Let's make it obvious in the doc string too.
```suggestion
    """This is a deprecated, internal-only class in ion-python—do NOT use it for any reason.

    Metaclass for simple enumerations.
```

**Comment by cheqianh** [comment]
The CI/CD failed due to C extension. which does not have any change. Looking into it right now.

**Comment by cheqianh** [comment]
Found the root, see - https://github.com/amazon-ion/ion-python/issues/249

We will skip these 10 unit tests now.

File: amazon/ion/util.py [hunk 3]
```diff
@@ -217,15 +226,106 @@ def bit_length(value):
...
+    _enum_members = {}
+
+    def __new__(cls, name, value):
+        return int.__new__(cls, value)
+
+    def __init__(self, name, value):
+        warn(f'{self.__class__.__name__} was deprecated in favor of the deprecation of `amazon.ion.Enum`. However, for '
+             f'compatibility reasons, `amazon.ion.Enum` has been placed here. We recommend considering using `IntEnum` '
+             f'as an alternative.',
+             DeprecationWarning, stacklevel=2)
```
[the remark this answers is not in the document]
Comment (cheqianh) [reply amazon/ion/util.py]: Agree, changed.

File: amazon/ion/util.py (hunk 1 above)
Comment (cheqianh) [reply amazon/ion/util.py]: Changed.

File: amazon/ion/util.py (hunk 2 above)
Comment (cheqianh) [reply amazon/ion/util.py]: Addressed.

**Comment by popematt** [comment]
Looks like the build is failing in the Python 3.10 workflow because of an error while trying to load the c extension. Do you know why that might be?

**Comment by cheqianh** [comment]
The CI/CD will fail for all python 3.x versions due to the recent ion-c release. I've navigated [the commit](https://github.com/amazon-ion/ion-c/commit/621e3796968170c520ea06a89b4baea26c43b070) that causes this change. Further investigation is required to identify the root cause. 

Reference issue - https://github.com/amazon-ion/ion-python/issues/249 for details.

File: ion-c
Comment (tgregg) [inline ion-c:1]: Let's revert the submodule update and address that in a separate PR, since it introduced errors that haven't yet been diagnosed.

**Comment by cheqianh** [comment]
The CI/CD passed due to the mitigation we had https://github.com/amazon-ion/ion-python/pull/250, which the commit will be reverted once the root cause is fixed.

**Review by popematt — APPROVED** [review]

**Pull request merged**